circumstellar-habitable-zone-simulator/src/diagram.jsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import * as PIXI from 'pixi.js';
+import PropTypes from 'prop-types';
+import STAR_SYSTEMS from './data';
+
+// Sun's diameter in pixels
+const AU_PIXELS = 100;
+const STAR_ORIGIN_POINT = [100, 150];
+const KM_AU = 149597870.7;
+const SOLAR_RADIUS_KM = 695700;
+
+const ZOOM_UPPER_BREAKPOINT = 960 * 0.8;
+const ZOOM_LOWER_BREAKPOINT = STAR_ORIGIN_POINT[0] + 20;
+
+
+export default class CSHZDiagram extends React.Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            zoomLevel: 0
+        }
+
+        this.cshzDiagram = React.createRef();
+        
+        this.renderStarSystem = this.renderStarSystem.bind(this);
+        this.renderStar = this.renderStar.bind(this);
```

Comment:
Normal instance methods that aren't used as callbacks don't need to be bound to `this`, do they?

Reply:
No, strictly speaking they don't need to be bound. I like to do it because I may update these methods in the future, and I don't want to remember if they're bound or not based on their usage. In this case though, `renderStar` does need access to `this`.